File: src/scripts/send.py
from time import sleep
import io
import serial
def main():
    ser = serial.Serial('/dev/ttyS8', 115200)
    with open('../img/sent.jpg', 'rb') as image:
        buffer_size = 32 #bytes
        while True:
            buffer = image.read(buffer_size)
            if not buffer: break
            ser.write(buffer)
            print(buffer)
            sleep(1)
if __name__ == '__main__':
    main()

File: src/scripts/send.cpp
#include <fstream>
#include <iostream>
#include <utility>
#include <string>
#include <array>
#include <vector>
int main() {
  std::ifstream image("../img/sent.jpg", std::ios::binary);
  std::vector<std::array<char, 64>> buffers;
  if (image.is_open()) {
    std::cout << image.rdbuf();
  }
  std::cout << std::endl;
  return 0;
}
